packages/babel-core/src/index.ts

```diff
@@ -92,8 +92,8 @@ export const DEFAULT_EXTENSIONS = Object.freeze([
   ".cjs",
 ] as const);
 
-import Module from "module";
-import * as thisFile from "./index.ts";
+import Module from "module" with { USE_ESM: "true", IS_STANDALONE: "false" };
+import * as thisFile from "./index.ts" with { USE_ESM: "true", IS_STANDALONE: "false" };
```

Comment:
This is creative! It seems to me that there is an implicit `&&` logic between multiple boolean toggles, while we could support `||` such as `{ or: { conditions } }`, I am wondering if we can use/abuse the dynamic `import()` for such purpose, the demonstrated example is `&&` but we can easily replace it with `||` if it is required:

```js
const thisFile = await import("./index.ts", { if: USE_ESM && !IS_STANDALONE });
```

If the variable declaration is at top level, we can just replace it by static import after evaluating the `if` option bag. The pros are support of more complicate conditions and the cons are that tools may not support dynamic import quite well as the static one.

Reply:
I'd prefer to keep them as import declarations also in the authored code, because top-level await has different semantics.

Wdyt about putting them in a string? i.e.
```js
import * as thisFile from "./index.ts" with { if: "USE_ESM && !IS_STANDALONE" };
```

Reply:
Understood. That looks good to me. 👍